src/components/InteractionLayer/index.js

```diff
@@ -190,8 +283,55 @@ class InteractionLayer extends React.Component {
     }
   };
 
+  // TODO: This extrapolate thing is super gross and so hacky.
```

Comment:
There are only a few, but I agree -- I just kept adding more until it worked, and then pared back all that we're absolutely necessary. Then I added the comment because I was ashamed.

Which one(s) do you think are unnecessary?